sphinx/application.py

```diff
@@ -1181,6 +1182,13 @@ def add_env_collector(self, collector: Type[EnvironmentCollector]) -> None:
         logger.debug('[app] adding environment collector: %r', collector)
         collector().enable(self)
 
+    def add_html_assets_in_all_pages(self):
```

Comment:
I'm thinking what is the best interface for this feature.

* "add" is not good for this case. The flag is not addable.
* "set" is better than "add".
* "html_assets_in_all_pages" flag is a bit many worded. A setting like "html_assets_policy=always" would be simpler than a flag.
* It might be better to add a method to set a setting for general purpose like `set_extension_variable('html', 'html_assets_in_all_pages', True)`

Reply:
I'd like to hear the opinion from @shimizukawa . So let's wait for him for a while.

Reply:
> "add" is not good for this case. The flag is not addable.
 
Yeah. I tried to follow the pattern that I saw. However, now that I'm reading it more carefully, `add_` is being used with a different meaning than I gave it to my variable.

> "set" is better than "add".

`set_` may works better, yes 👍🏼 

> "html_assets_in_all_pages" flag is a bit many worded. A setting like "html_assets_policy=always" would be simpler than a flag.

IMO, I would say that it makes sense to use this approach if we already know that we will have other different policies --do we?. Otherwise, I think we will be just complicating the API.

> It might be better to add a method to set a setting for general purpose like `set_extension_variable('html', 'html_assets_in_all_pages', True)`

I like the "general purpose" solution. I'd expect to have more of these extension variables as Sphinx keeps growing. However, we still need to define a good name for the variable, right?

Reply:
+1 for `set_`.

`html_assets_policy='always'` is fine to me because:

* Shorter name
* Potential for future enhancement

Reply:
@shimizukawa Thank you for your comment. Okay, let's go with `set_html_assets_policy('always')` (The default is `'per_page'`).